std/string.d

```diff
@@ -871,20 +871,44 @@ private template _indexOfStr(CaseSensitive cs)
         Does not work with case insensitive strings where the mapping of
         tolower and toupper is not 1:1.
   +/
-ptrdiff_t indexOf(Range, Char)(Range s, const(Char)[] sub,
-        in CaseSensitive cs = Yes.caseSensitive)
+ptrdiff_t indexOf(Range, Char)(Range s, const(Char)[] sub)
+if (isForwardRange!Range && isSomeChar!(ElementEncodingType!Range) &&
+    isSomeChar!Char)
+{
+    return _indexOfStr!(Yes.caseSensitive)(s, sub);
+}
+
+/// Ditto
+ptrdiff_t indexOf(Range, Char)(Range s, const(Char)[] sub, in CaseSensitive cs)
 if (isForwardRange!Range && isSomeChar!(ElementEncodingType!Range) &&
     isSomeChar!Char)
 {
     if (cs == Yes.caseSensitive)
-        return _indexOfStr!(Yes.caseSensitive)(s, sub);
+        return indexOf(s, sub);
     else
         return _indexOfStr!(No.caseSensitive)(s, sub);
 }
 
 /// Ditto
 ptrdiff_t indexOf(Char1, Char2)(const(Char1)[] s, const(Char2)[] sub,
-        in size_t startIdx, in CaseSensitive cs = Yes.caseSensitive)
+        in size_t startIdx)
+@safe
+if (isSomeChar!Char1 && isSomeChar!Char2)
+{
+    if (startIdx < s.length)
+    {
+        ptrdiff_t foundIdx = indexOf(s[startIdx .. $], sub);
+        if (foundIdx != -1)
+        {
+            return foundIdx + cast(ptrdiff_t) startIdx;
```

Comment:
you cloud do foundIdx += cast(ptrdiff_t) startIdx;

and then return foundIdx at the end, which would safe one ret instruction and some branch predictor pain,

Reply:
Also, this is not my code. The function was duplicated in order to allow it to have its own inferred attributes, due to case sensitivity no longer being variable at runtime.

Reply:
Hmm dmd amazes es always :)

Reply:
my suggestion was different though
```
    ptrdiff_t foundIdx = -1;
    if (startIdx < s.length)
    {
        foundIdx = indexOf(s[startIdx .. $], sub);
        if (foundIdx != -1)
        {
            foundIdx += cast(ptrdiff_t) startIdx;
        }    
    }
    return foundIdx;
```

Reply:
Tried that too, still more instructions...